tests/unit/test_client.py

```diff
@@ -9546,6 +9551,37 @@ def test_schema_to_json_with_file_object(self):
         client.schema_to_json(schema_list, fake_file)
         assert file_content == json.loads(fake_file.getvalue())
 
+    if sys.version_info >= (3, 9):
+
+        @patch("bigquery_client.async_query_and_wait", new_callable=AsyncMock)
+        async def test_async_execution(self, mock_query):
```

Comment:
Are we sure this test is detected and runnable by pytest? From https://stackoverflow.com/a/70016047/101923 it seems we may need to install another plugin for pytest and add a decorator.

Reply:
I did run the test without the if statement, it understandably caused errors for 3.7/3.8 for not being supported however passed for other python versions. It seems like the proof of concept works, but I will revisit how to run this test according to this example in the next commit!

Reply:
Because I am using asyncio.to_thread, should I test if it returns a coroutine before the await, instead of a future?